src/test/java/org/mockitousage/matchers/InvalidUseOfMatchersTest.java

```diff
@@ -93,4 +98,35 @@ public void should_scream_when_Matchers_count_dont_match_parameter_count() {
                     .contains("1 recorded");
         }
     }
+
+    @Test
+    public void should_mention_matcher_when_misuse_detected() {
+        // Given
+
+
+        // When
+        Result run = new JUnitCore().run(ObjectMatcherMisuseOnPrimitiveSite.class);
+
+        // Then
+
+        assertThat(run.getFailures()).hasSize(2);
+        assertThat(run.getFailures().get(0).getException()).isInstanceOf(NullPointerException.class)
+                                                           .hasMessage(null);
+        assertThat(run.getFailures().get(1).getException()).isInstanceOf(InvalidUseOfMatchersException.class)
+                                                           .hasMessageContaining("primitive alternatives");
+        new StateMaster().reset();
+
+    }
+
+    @RunWith(MockitoJUnitRunner.class)
+    public static class ObjectMatcherMisuseOnPrimitiveSite {
+        @Test
+        public void fails_with_NPE() {
+            IMethods mock = Mockito.mock(IMethods.class);
+            doNothing().when(mock)
+                       .twoArgumentMethod(eq(73),
+                                          (Integer) any()); // <= Raise NPE on this call site
```

Comment:
Oof indentation looks weird on this one. Nit: move this to the previous line for readibility